src/main/java/io/kestra/plugin/meilisearch/FacetSearch.java

```diff
@@ -0,0 +1,123 @@
+        WARNING: make sure to set the [filterable attributes](https://www.meilisearch.com/docs/learn/filtering_and_sorting/search_with_facet_filters#configure-facet-index-settings) before.
+        """
+)
+@Plugin(
+    examples = {
+        @io.kestra.core.models.annotations.Example(
+            title = "Sample facet search",
+            code = {
+                """
+                    facetQuery: "fiction",
+                    facetName: "genre",
+                    filters:
+                        -"rating > 3"
+                    url: "http://localhost:7700",
+                    key: "MASTER_KEY",
+                    index: "movies"
+
+                    id: meilisearch-facet-search-flow
+                        namespace: company.team
+
+                    variables:
+                      index: movies
+                      facetQuery: fiction
+                      facetName: genre
+                      host: http://172.18.0.3:7700/
+
+                    tasks:
+                      - id: facet_search_documents
+                        type: io.kestra.plugin.meilisearch.FacetSearch
+                        index: {{ vars.index }}
+                        facetQuery: {{ vars.facetQuery }}
+                        facetName: {{ vars.facetName }}
+                        filters:
+                            - "rating > 3"
+                        url: "{{ vars.host }}"
+                        key: "MASTER_KEY"
+
+                      - id: to_json
+                        type: io.kestra.plugin.serdes.json.IonToJson
+                        from: "{{ outputs.search_documents.uri }}"
+                """
+            }
+        )
+    }
+)
+public class FacetSearch extends AbstractMeilisearchConnection implements RunnableTask<FacetSearch.Output> {
+    @PluginProperty(dynamic = true)
+    private Property<String> index;
+
+    @PluginProperty(dynamic = true)
+    private Property<String> facetName;
+
+    @PluginProperty(dynamic = true)
+    private Property<String> facetQuery;
+
+    @PluginProperty(dynamic = true)
+    private List<String> filters;
+
+    @Override
+    public FacetSearch.Output run(RunContext runContext) throws Exception {
+        Client client = this.createClient(runContext);
+        Index searchIndex = client.index(index.as(runContext, String.class));
+
+        FacetSearchRequest fsr = FacetSearchRequest.builder()
+            .facetName(facetName.as(runContext, String.class))
+            .facetQuery(facetQuery.as(runContext, String.class))
+            .filter(runContext.render(filters).toArray(new String[]{}))
+            .build();
+
+        var results = searchIndex.facetSearch(fsr);
+
+        File tempFile = runContext.workingDir().createTempFile(".ion").toFile();
+        try (var output = new BufferedWriter(new FileWriter(tempFile), FileSerde.BUFFER_SIZE)) {
+            Flux<FacetSearchable> hitFlux = Flux.just(results);
```

Comment:
```suggestion
            Flux<FacetSearchable> hitFlux = Flux.fromIterable(results);
```

Flux.just() will emit a single element, so if you have a list you want to emit multiple elements so each element will be serialized as a separate object so each element will be in its own line
Here, you serialize all elements into a big array into a single line.

Reply:
Actually the result returned by facetSearch is always one JSON document like below :  
That's why I just put `Flux.just`. But I will rename `results` to `result`.

Example: 
`{
  "facetHits":[
    {
      "value":"fiction",
      "count":7
    }
  ],
  "facetQuery":"fiction",
  "processingTimeMs":0
}`